src/bin/miri.rs

```diff
@@ -560,16 +560,17 @@ fn main() {
                 }));
             miri_config.provenance_mode = ProvenanceMode::Strict;
         } else if arg == "-Zmiri-tree-borrows-no-precise-interior-mut" {
-            if matches!(miri_config.borrow_tracker, Some(BorrowTrackerMethod::TreeBorrows { .. })) {
-                miri_config.borrow_tracker =
-                    Some(BorrowTrackerMethod::TreeBorrows(TreeBorrowsParams {
-                        precise_interior_mut: false,
-                    }));
-            } else {
-                show_error!(
-                    "`-Zmiri-tree-borrows` must also be set before the `-Zmiri-tree-borrows-no-precise-interior-mut` flag"
-                );
-            }
+            miri_config.borrow_tracker.as_mut().map(|method| {
```

Comment:
This means there'll be no error if someone does `-Zmiri-disable-stacked-borrows `-Zmiri-tree-borrows-no-precise-interior-mut`."

You'll have to do something like
```rust
match &mut miri_config.borrow_tracker {
  BorrowTrackerMethod::TreeBorrows(ref mut params) => ...
  _ => { /* show errror */ }
}
```